Net/Gearman/Client.php

```diff
@@ -323,6 +323,37 @@ public function runSet(Net_Gearman_Set $set, $timeout = null)
         }
     }
 
+    /**
+     * Get the status of a task.
+     *
+     * @param $handle string The task handle returned when the task is submitted.
+     * @return array|false Information about the task.
+     */
+    public function getStatus($handle)
+    {
+        $conn = $this->getConnection();
+
+        $conn->send('get_status', compact('handle'));
+
+        $read = $write = $except = [];
+
+        foreach ($this->conn as $conn) {
+            $read[] = $conn->socket;
+        }
+
+        socket_select($read, $write, $except, 10);
```

Comment:
You likely want more error checking following this call. Silent socket errors can lead to hard to find bugs. And socket_select can be noisy. See the other usage in this file in `runSet`. That block could be abstracted into its own function and used by both this function and runSet.